it/src/test/java/com/google/cloud/teleport/it/spanner/DefaultSpannerResourceManagerTest.java

```diff
@@ -0,0 +1,364 @@
+                .set("FirstName").to("Catalina")
+                .set("LastName").to("Smith")
+                .build());
+    // spotless:on
+
+    // act & assert
+    assertThrows(SpannerResourceManagerException.class, () -> testManager.write(testMutations));
+  }
+
+  @Test
+  public void testReadRecordsShouldWorkWhenSpannerReadSucceeds()
+      throws ExecutionException, InterruptedException {
+    // arrange
+    prepareTable();
+    when(resultSet.next()).thenReturn(true).thenReturn(true).thenReturn(false);
+    // spotless:off
+    Struct struct1 =
+        Struct.newBuilder()
+            .set("SingerId").to(int64(1))
+            .set("FirstName").to(string("Marc"))
+            .set("LastName").to(string("Richards"))
+            .build();
+    Struct struct2 =
+        Struct.newBuilder()
+            .set("SingerId").to(int64(2))
+            .set("FirstName").to(string("Catalina"))
+            .set("LastName").to(string("Smith"))
+            .build();
+    // spotless:on
+    when(resultSet.getCurrentRowAsStruct()).thenReturn(struct1).thenReturn(struct2);
+    when(spanner.getDatabaseClient(any()).singleUse().read(any(), any(), any()))
+        .thenReturn(resultSet);
+
+    // act
+    ImmutableList<Struct> actual =
+        testManager.readTableRecords("Singers", "SingerId", "FirstName", "LastName");
+
+    // assert
+    ImmutableList<Struct> expected = ImmutableList.of(struct1, struct2);
+    assertThat(actual).containsExactlyElementsIn(expected);
+  }
+
+  @Test
+  public void testReadRecordsShouldThrowExceptionWhenSpannerReadFails()
+      throws ExecutionException, InterruptedException {
+    // arrange
+    prepareTable();
+    when(spanner.getDatabaseClient(any()).singleUse().read(any(), any(), any()))
+        .thenThrow(SpannerException.class);
+
+    // act & assert
+    assertThrows(
+        SpannerResourceManagerException.class,
+        () -> testManager.readTableRecords("Singers", "SingerId", "FirstName", "LastName"));
+  }
+
+  @Test
+  public void testCleanupAllShouldThrowExceptionWhenSpannerDeleteInstanceFails() {
+    // arrange
+    doThrow(SpannerException.class).when(instanceAdminClient).deleteInstance(any());
+    when(spanner.getInstanceAdminClient()).thenReturn(instanceAdminClient);
+    testManager = new DefaultSpannerResourceManager(spanner, TEST_ID, PROJECT_ID, REGION);
+
+    // act & assert
+    assertThrows(SpannerResourceManagerException.class, () -> testManager.cleanupAll());
+  }
+
+  @Test
+  public void testCleanupAllShouldWorkWhenSpannerDeleteInstanceSucceeds() {
+    // arrange
+    doNothing().when(instanceAdminClient).deleteInstance(any());
+    when(spanner.getInstanceAdminClient()).thenReturn(instanceAdminClient);
+    testManager = new DefaultSpannerResourceManager(spanner, TEST_ID, PROJECT_ID, REGION);
+
+    // act
+    testManager.cleanupAll();
+
+    // assert
+    verify(spanner.getInstanceAdminClient()).deleteInstance(any());
+    verify(spanner).close();
+  }
+}
```

Comment:
Sorry, missed this last time. Could we add tests for what happens when we try to use this after `cleanupAll` is called?